qubes/app.py

```diff
@@ -1251,6 +1251,10 @@ def _domain_event_callback(self, _conn, domain, event, _detail, _opaque):
 
         if event == libvirt.VIR_DOMAIN_EVENT_STOPPED:
             vm.on_libvirt_domain_stopped()
+        elif event == libvirt.VIR_DOMAIN_EVENT_SUSPENDED:
+            vm.fire_event('domain-paused')
+        elif event == libvirt.VIR_DOMAIN_EVENT_RESUMED:
+            vm.fire_event('domain-unpaused')
```

Comment:
Does this synchronously call all event handlers in this thread context? Or just enqueue them for the run loop. If the former, should I be catching here to protect libvirt?